plugins/sonarqube-backend/src/service/router.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { errorHandler } from '@backstage/backend-common';
+import express, { RequestHandler } from 'express';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+import {
+  SonarqubeFindings,
+  SonarqubeInfoProvider,
+} from './sonarqubeInfoProvider';
+import { InputError } from '../../../../packages/errors';
+
+/**
+ * Dependencies needed by the router
+ * @public
+ */
+export interface RouterOptions {
+  /**
+   * Logger for logging purposes
+   */
+  logger: Logger;
+  /**
+   * Info provider to be able to get all necessary information for the APIs
+   */
+  sonarqubeInfoProvider: SonarqubeInfoProvider;
+}
+
+/**
+ * @public
+ *
+ * Constructs a sonarqube router.
+ *
+ * Expose endpoint to get information on or for a sonarqube instance.
+ *
+ * @param options - Dependencies of the router
+ */
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const { logger, sonarqubeInfoProvider } = options;
+
+  const router = Router();
+  router.use(express.json());
+  // mock api for now
+  router.get('/findings', (async (request, response) => {
+    const componentKey = request.query.componentKey;
+    let instanceKey = request.query.instanceKey;
+
+    if (!componentKey)
+      throw new InputError('ComponentKey must be provided as a single string.');
+
+    if (!instanceKey) {
+      instanceKey = '';
+      logger.info(
+        `Retrieving findings for component ${componentKey} in default sonarqube instance`,
+      );
+    } else {
+      logger.info(
+        `Retrieving findings for component ${componentKey}  in sonarqube instance name ${instanceKey}`,
+      );
+    }
+
+    response.send(
+      await sonarqubeInfoProvider.getFindings(componentKey, instanceKey),
+    );
+  }) as RequestHandler<unknown, SonarqubeFindings | undefined, unknown, { componentKey: string; instanceKey: string }>);
```

Comment:
Is this just so you can type the query? It's a little unclear.

I think you can just do `const componetnKey = request.query.componentKey as string`

Reply:
Yeah, I'm wondering maybe we look at making defining this contract a little easier in some of the upcoming `backend-system` work we're doing.

Basically being able to register routes and define the types that you expect using something like `zod` and being able to do validation for you outside of the handler. 

I think let's keep it simple for now, and have it in mind for later.